### ISSUE TITLE
How can I get the instance-wise gradient?

### ISSUE THREAD

**Issue by JiahaoYao** [tensorflow/tensorflow#29926]
**System information**
- TensorFlow version (you are using): Tensorflow 1.1.13

**Describe the feature and the current behavior/state.**

Hi, Does anyone know whether tensorflow could provide instance-wise gradient? To be more precise, I mean say the loss is not a scalar, just the size of (N, 1). This is cross entropy loss before one use `tf.reduce_mean`. I wonder how could I get the gradient for each instance in each batch. This means eventually, the shape of the gradient is supposed to be (N, num_of_thetas). I think the currently `tf.gradient` can not do this, right?

**Comment by dynamicwebpaige** [comment]
@JiahaoYao - This is a support question, and would be better suited for StackOverflow.

The gradient terms for each tensor are automatically aggregated. Gradient computation operations such as `optimizer.compute_gradients` and the low-level primitive [`tf.gradients`](https://www.tensorflow.org/api_docs/python/tf/gradients) make a sum of all gradient operations, according to the default `AddN` aggregation method. This is fine for most cases of stochastic gradient descent.

In the end unfortunately, gradient computation will have to be made over a single batch. Of course, unless a custom gradient function is built, or the TensorFlow API is extended to provide gradient computation without full aggregation.  [Link to implementation of `tf.gradients`.](https://github.com/tensorflow/tensorflow/blob/r1.2/tensorflow/python/ops/gradients_impl.py#L368)

**Comment by JiahaoYao** [comment]
Thanks @dynamicwebpaige !
